jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,821 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cubeviz_data = cubeviz.app.data_collection[0]\n",
+    "try:\n",
+    "    #region1=cubeviz.app.get_subsets_from_viewer(\"spectrum-viewer\")\n",
+    "    #print(region1['Subset 1'])\n",
+    "\n",
+    "    region1=cubeviz_data.get_selection_definition(format='astropy-regions')\n",
+    "    print(region1)\n",
+    "    region1_exists=True\n",
+    "except:\n",
+    "    print('There are no regions selected in the cube viewer.')\n",
+    "    region1_exists=False\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extract Spectrum using Subset in Cubeviz Spectrum Viewer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    spectrum_subset1=cubeviz.app.get_data_from_viewer(\"spectrum-viewer\")['Subset 1']\n",
+    "    print(spectrum_subset1)\n",
+    "except:\n",
+    "    print('There are no subsets selected in the spectrum viewer.')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Can we suppress or fix this glue/core warning?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extract Spectrum by Sum Over Spaxels\n",
+    "\n",
+    "Perform a simple numpy sum over all spaxels in the cube. The resultant spectra show broad and narrow emission lines from the quasar (H-beta, [O III], H-alpha, etc..).  We also extract all of the spatial slices into a list to prepare them for aperture photometry."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Sum over wavelength\n",
+    "cube_sum=np.sum(data_win,axis=0)\n",
+    "\n",
+    "##Plots\n",
+    "f,(ax1,ax2) = plt.subplots(1,2,figsize=(10, 5)) \n",
+    "ax1.plot(wavelength, fnu_sum) \n",
+    "ax1.set_xlim(0.95,1.5)\n",
+    "ax1.set_title('Spaxel sums')\n",
+    "ax1.set_xlabel(\"Wavelength (um)\")  \n",
+    "ax1.set_ylabel(\"SCA491 Flux Density\")\n",
+    "\n",
+    "ax2.imshow(cube_sum,norm=LogNorm())\n",
+    "ax2.set_title('Slice sums')\n",
+    "\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extract Spectrum in Constant Radius Circular Aperture (Cylinder)\n",
+    "This method is appropriate for an extended source."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#IFU pixel scale\n",
+    "pixelscale=0.1\n",
+    "\n",
+    "#Circular Extraction region, centered on QSO\n",
+    "#Use this hard-coded aperture or the aperture retrieved from Cubeviz\n",
+    "coords = \"10:24:42.677 -45:41:13.041\"\n",
+    "center= SkyCoord(coords,unit=(u.hourangle, u.deg), frame=FK5)\n",
+    "rarcs=0.592\n",
+    "\n",
```

Comment:
The region exported from CUbeviz is given in pixels

Region: CirclePixelRegion
center: PixCoord(x=17.116783142089844, y=21.200183868408203)
radius: 3.710627555847168

Can these numbers be used instead of the hard-coded aperture ?

Reply:
Yes, I will try to do this.